crates/net/discv4/src/node.rs

```diff
@@ -38,246 +25,8 @@ pub(crate) fn kad_key(node: PeerId) -> discv5::Key<NodeKey> {
-            })
-        }
-        let rem = rlp_head.payload_length - consumed;
-        b.advance(rem);
-        *buf = *b;
-        Ok(this)
-    }
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use bytes::BytesMut;
-    use rand::{thread_rng, Rng, RngCore};
-
-    #[test]
-    fn test_noderecord_codec_ipv4() {
-        let mut rng = thread_rng();
-        for _ in 0..100 {
-            let mut ip = [0u8; 4];
-            rng.fill_bytes(&mut ip);
-            let record = NodeRecord {
-                address: IpAddr::V4(ip.into()),
-                tcp_port: rng.gen(),
-                udp_port: rng.gen(),
-                id: PeerId::random(),
-            };
-
-            let mut buf = BytesMut::new();
-            record.encode(&mut buf);
-
-            let decoded = NodeRecord::decode(&mut buf.as_ref()).unwrap();
-            assert_eq!(record, decoded);
-        }
-    }
-
-    #[test]
-    fn test_noderecord_codec_ipv6() {
-        let mut rng = thread_rng();
-        for _ in 0..100 {
-            let mut ip = [0u8; 16];
-            rng.fill_bytes(&mut ip);
-            let record = NodeRecord {
-                address: IpAddr::V6(ip.into()),
-                tcp_port: rng.gen(),
-                udp_port: rng.gen(),
-                id: PeerId::random(),
-            };
-
-            let mut buf = BytesMut::new();
-            record.encode(&mut buf);
-
-            let decoded = NodeRecord::decode(&mut buf.as_ref()).unwrap();
-            assert_eq!(record, decoded);
-        }
-    }
-
-    #[test]
-    fn test_url_parse() {
-        let url = "enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@10.3.58.6:30303?discport=30301";
-        let node: NodeRecord = url.parse().unwrap();
-        assert_eq!(node, NodeRecord {
-            address: IpAddr::V4([10,3,58,6].into()),
-            tcp_port: 30303,
-            udp_port: 30301,
-            id: "6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0".parse().unwrap(),
-        })
-    }
-
-    #[test]
-    fn test_node_display() {
-        let url = "enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@10.3.58.6:30303";
-        let node: NodeRecord = url.parse().unwrap();
-        assert_eq!(url, &format!("{node}"));
-    }
-
-    #[test]
-    fn test_node_display_discport() {
-        let url = "enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@10.3.58.6:30303?discport=30301";
-        let node: NodeRecord = url.parse().unwrap();
-        assert_eq!(url, &format!("{node}"));
-    }
+/// Returns the key type for the kademlia table
+#[inline]
+pub(crate) fn record_key(node: &NodeRecord) -> discv5::Key<NodeKey> {
```

Comment:
Any reason why we can't do `From<&NodeRecord>` here? Is it to avoid the discv5 dependency on primitives?

Reply:
yes, to avoid discv5 dep in primitives

Reply:
Yes we can! Thanks for the idea! I'm changing this to a `From<&NodeRecord>` trait impl.